[torch][fx] Add support for EXIR dialect overload ops in normalize_function

dulinriley opened the pull request:
Summary:
I had a minor annoyance when debugging graphs using EXIR dialect ops,
that all the function normalization went away. For functions with > 5 arguments,
some of which are just simple bools and ints, it's very helpful to have
the kwarg names attached.

Enhance `normalize_target` to handle EdgeOpOverload targets. To avoid
a circular dependency on Executorch from pytorch core, I just use a `hasattr`
check for "_op". This only happens if the target is not already a recognized
torch function.

Also, I noticed that the new `fx.Node.normalized_arguments` function
didn't forward an important kwarg to `normalize_target`, so I fixed that too.

Test Plan: Tested with FxGraphDrawer and an fx Graph containing EXIR nodes.

Differential Revision: D67545909




cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv

angelayi commented on torch/fx/operator_schemas.py:
```diff
@@ -366,6 +366,16 @@ def normalize_function(
+    if not isinstance(target, types.BuiltinFunctionType) and not (
+        isinstance(target, (OpOverloadPacket, OpOverload))
+    ) and hasattr(target, "_op"):
+        # The Executorch EXIR dialect targets have an underlying OpOverload which
+        # should be used for this function.
+        # Can't import EdgeOpOverload directly because of a circular dependency,
+        # so checking for "_op" existing is the next best thing.
```
Can you edit this to say something like, "ExecuTorch's EdgeOpOverload are a wrapper around PyTorch's OpOverload, so we can unwrap it here to get its schema"

dulinriley replied:
Oh my bad I hadn't been monitoring the PR for a while, just the diff. Let me refresh this and address your comment

dulinriley replied:
Applied this comment to a new version

dulinriley committed:
[torch][fx] Add support for EXIR dialect overload ops in normalize_function (#143689)

Summary:

I had a minor annoyance when debugging graphs using EXIR dialect ops,
that all the function normalization went away. For functions with > 5 arguments,
some of which are just simple bools and ints, it's very helpful to have
the kwarg names attached.

Enhance `normalize_target` to handle EdgeOpOverload targets. To avoid
a circular dependency on Executorch from pytorch core, I just use a `hasattr`
check for "_op". This only happens if the target is not already a recognized
torch function.

Also, I noticed that the new `fx.Node.normalized_arguments` function
didn't forward an important kwarg to `normalize_target`, so I fixed that too.

Test Plan: Tested with FxGraphDrawer and an fx Graph containing EXIR nodes.

Reviewed By: angelayi

Differential Revision: D67545909
